cmd/moq-example/client.cpp

```diff
@@ -0,0 +1,396 @@
+    }
+    void cb_clientSetup(qtransport::TransportConnId conn_id, quicr::messages::MoqClientSetup client_setup) override {}
+    void cb_serverSetup(qtransport::TransportConnId conn_id, quicr::messages::MoqServerSetup server_setup) override {}
+
+private:
+    cantina::LoggerPointer _logger;
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * Publisher Thread to perform publishing
+ * -------------------------------------------------------------------------------------------------
+ */
+void do_publisher(const std::string t_namespace,
+                  const std::string t_name,
+                  const std::shared_ptr<quicr::MoQInstance>& moqInstance,
+                  const cantina::LoggerPointer& logger,
+                  const bool& stop)
+{
+    cantina::LoggerPointer _logger = std::make_shared<cantina::Logger>("PUB", logger);
+
+    auto mi = moqInstance;
+
+    auto track_delegate = std::make_shared<trackDelegate>(t_namespace, t_name, 2, 3000, logger);
+
+    _logger->info << "Started publisher track: " << t_namespace << "/" << t_name << std::flush;
+
+    bool published_track { false };
+    bool sending { false };
+    uint64_t group_id { 0 };
+    uint64_t object_id { 0 };
+
+    while (not stop) {
+        if (!published_track && qclient_vars::conn_id) {
+            _logger->info << "Publish track: " << t_namespace << "/" << t_name << std::flush;
+            mi->publishTrack(*qclient_vars::conn_id, track_delegate);
+            published_track = true;
+        }
+
+        if (track_delegate->statusSend() != quicr::MoQTrackDelegate::TrackSendStatus::OK) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+            continue;
+        }
+
+        if (!sending) {
+            _logger->info << "--------------------------------------------------------------------------" << std::flush;
+
+            if (qclient_vars::publish_clock) {
+                _logger->info << " Publishing clock timestamp every second" << std::flush;
+            } else {
+                _logger->info << " Type message and press enter to send" << std::flush;
+            }
+
+            _logger->info << "--------------------------------------------------------------------------" << std::flush;
+            sending = true;
+        }
+
+        std::string msg;
+        if (qclient_vars::publish_clock) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(999));
+            msg = get_time_str();
+            _logger->info << msg << std::flush;
+        } else { // stdin
+            getline(std::cin, msg);
+            _logger->info << "Send message: " << msg << std::flush;
+        }
+
+        if (object_id % 5 == 0) {       // Set new group
+            object_id = 0;
+            group_id++;
+        }
+
+        std::vector<uint8_t> msg_v(msg.begin(), msg.end());
+        auto result = track_delegate->sendObject(group_id, object_id++, msg_v);
```

Comment:
It is possible to avoid copying the string here by interpreting the string as a byte*:
```cpp
auto result = track_delegate->sendObject(group_id, object_id++, std::span{std::bit_cast<uint8_t*>(&msg), sizeof(std::string)});
```
Probably a good idea to check that cast, but that's the general idea.

Reply:
Should be fine to do that. Just updated.